src/raftstore/store/peer_storage.rs

```diff
@@ -1137,14 +1164,20 @@ impl PeerStorage {
     }
 
     /// Update the memory state after ready changes are flushed to disk successfully.
-    pub fn post_ready(&mut self, ctx: InvokeContext) -> Option<ApplySnapResult> {
+    pub fn post_ready(
+        &mut self,
+        ctx: InvokeContext,
+    ) -> (Option<ApplySnapResult>, Option<GenSnapTask>) {
+        // The snapshot gen task need to be sent the apply system.
+        let gen_snap = self.snap_gen_task.borrow_mut().take();
```

Comment:
What if the region hibernate? Followers can request snapshots even if there are no proposals for the region.

Reply:
I think it will be invoked as long as follower requests snapshot via raft message.